src/integration-test/java/com/commercetools/sync/integration/services/impl/UnresolvedReferencesServiceImplIT.java

```diff
@@ -92,6 +93,44 @@ void save_WithoutException_createsNewCustomObject() {
         assertThat(createdCustomObject.getValue()).isEqualTo(productDraftWithUnresolvedRefs);
     }
 
+    @Test
+    void save_ExistingProductDraftWithoutException_overwriteOldCustomObjectValue() {
+        // preparation
+        final ProductDraft productDraft =
+            SphereJsonUtils.readObjectFromResource(PRODUCT_KEY_1_RESOURCE_PATH, ProductDraft.class);
+
+        final WaitingToBeResolved productDraftWithUnresolvedRefs =
+                new WaitingToBeResolved(productDraft, asSet("foo", "bar"));
+
+        unresolvedReferencesService
+                .save(productDraftWithUnresolvedRefs)
+                .toCompletableFuture()
+                .join();
+
+        final WaitingToBeResolved productDraftWithUnresolvedNewRefs =
+                new WaitingToBeResolved(productDraft, asSet("foo123", "bar123"));
+
+        // test
+        final Optional<WaitingToBeResolved> latestResult = unresolvedReferencesService
+            .save(productDraftWithUnresolvedNewRefs)
+            .toCompletableFuture()
+            .join();
+
+
+        // assertions
+        assertThat(latestResult).hasValueSatisfying(waitingToBeResolved ->
+            assertThat(waitingToBeResolved.getProductDraft()).isEqualTo(productDraft));
+
+        final CustomObjectByKeyGet<WaitingToBeResolved> customObjectByKeyGet = CustomObjectByKeyGet
+            .of(CUSTOM_OBJECT_CONTAINER_KEY, productDraft.getKey(), WaitingToBeResolved.class);
+        final CustomObject<WaitingToBeResolved> createdCustomObject = CTP_TARGET_CLIENT
```

Comment:
It is to assert the entire `WaitingToBeResolved` object has changed to `productDraftWithUnresolvedNewRefs` in the CTP project. Basically this assert whether the old value has been replaced with new value. This test case is made because we do not maintain versioned custom objects.